Modules/Challenges/test/SpecFlowChallenges.Specs/Models/AuthenticationParams.cs

```diff
@@ -0,0 +1,10 @@
+﻿namespace SpecFlowChallenges.Specs.Models;
+public class AuthenticationParameters
+{
+    public Dictionary<string, string> Parameters { get; set; }
```

Comment:
My intention here was to make the code more dynamic since we can later add all the parameters to the HTTP request in one go, instead of multiple lines of code to add each parameter.

Reply:
I'm generally a fan of generic/dynamic code. And I even tend to drive it too far in that direction. That's why I keep asking myself "Is it really necessary to be THAT generic?". Most of the time the answer is "No". In this case (and many other cases as well), being generic comes with the cost of loosing readability. In this case, we also loose type safety, because a user of the AuthenticationParameters could by accident add a "userName" parameter instead of "username".

Also, always think about the questions "How often do I need to change the code?" and "What are the cost of changing the code?". 
In your case: 
* How often do you need to add a authentication parameter? Since OAuth is a widely used standard, you can probably expect there to be no changes at all. 
* What are the cost of changing the code? Well, you need to add a property to the class. If you declare it as `required`, the compiler tells you all the places in which it is missing. All in all, it's probably a matter of a few minutes.